content/cucumber/step-definitions.md

```diff
@@ -1,154 +1,170 @@
+When a Step Definition's method or function invokes the `pending` method, the Step is marked as yellow (as with `undefined` ones), indicating that you have work to do. If you use `--strict`, this will cause Cucumber to exit with `1`.
 
-## Failed Steps
+### Failed Steps
 
-When a Step Definition's Proc is executed and raises an error, the step is marked as red. What you return from a Step Definition has no significance whatsoever. 
+When a Step Definition's method or function is executed and raises an error, the step is marked as red. What you return from a Step Definition has no significance whatsoever.
 
-Returning `nil` or `false` will **not** cause a Step Definition to fail.
+Returning {{% text "ruby" %}}`nil`{{% /text %}}{{% text "java" %}}`null`{{% /text %}}{{% text "javascript" %}}`null`{{% /text %}} or `false` will **not** cause a Step Definition to fail.
 
-## Skipped Steps
+### Skipped Steps
 
 Steps that follow `undefined`, `pending`, or `failed` Steps are never executed,  even if there is a matching Step Definition. These Steps are marked as cyan.
 
-## String Steps
+{{% text "ruby" %}}
+### String Steps (Ruby)
 
-Step Definitions can be written using strings rather than regular expressions. 
+In Ruby, Step Definitions can be written using Strings rather than Regular Expressions.
 
 Instead of writing:
 
-```
+```ruby
 Given /^I have (.*) cucumbers in my belly$/ do |cukes|
 ```
 
 You could write:
 
-```
+```ruby
 Given "I have $count cucumbers in my belly" do |cukes|
 ```
 
-When writing a Step Definition using the string form, any word preceded by a `$` is taken to be a placeholder. Behind the scenes, Cucumber will convert it to the regular expression `(.*)`. 
+When writing a Step Definition using the String form, any word preceded by a `$` is taken to be a placeholder. Behind the scenes, Cucumber will convert it to the Regular Expression `(.*)`.
 
 The text matched by the wildcard becomes an argument to the block, and the word that appeared in the Step Definition is disregarded.
+{{% /text %}}
 
+### Ambiguous Steps
 
-## Ambiguous Steps
-
-Consider these Step Definitions:
-
-```
-Given /Three (.*) mice/ do |disability|
-  # some code
-end
-
-Given /Three blind (.*)/ do |animal|
-  # some other code..
-end
-```
-
-And a plain text step:
-
-```
-Given Three blind mice
-```
-
-Cucumber can't make a decision about what Step Definition to execute, and will raise a `Cucumber::Ambiguous` error telling you to fix the ambiguity.
-
+Step Definitions have to be unique for Cucumber to know what to execute.
+If you use ambiguous Step Definitions,{{% text "ruby" %}}Cucumber will raise a `Cucumber::Ambiguous` error,{{% /text %}}
+{{% text "java" %}} Cucumber will raise an `AmbiguousStepDefinitionsException`,{{% /text %}}
+{{% text "javascript" %}}the Step / Scenario will get an "Ambiguous" result,{{% /text %}}
+telling you to fix the ambiguity.
 
+{{% text "ruby" %}}
```

Comment:
I think this one is surplus?

Reply:
Actually, I think this may be Ruby specific which is why I tagged these 2 paragraphs as "Ruby". If I'm mistaken, it should be updated.